src/NServiceBus.AcceptanceTests/Core/OpenTelemetry/Metrics/TestingMetricListener.cs

```diff
@@ -36,11 +36,21 @@ public TestingMetricListener(string sourceName)
             ReportedMeters.AddOrUpdate(instrument.Name, measurement, (_, val) => val + measurement);
             Tags.AddOrUpdate(instrument.Name, _ => tags, (_, _) => tags);
         });
+        meterListener.SetMeasurementEventCallback((Instrument instrument,
+            double measurement,
+            ReadOnlySpan<KeyValuePair<string, object>> t,
+            object _) =>
+        {
+            TestContext.WriteLine($"{instrument.Meter.Name}\\{instrument.Name}:{measurement}");
+            var tags = t.ToArray();
+            ReportedMeters.AddOrUpdate(instrument.Name, 1, (_, val) => val + 1);
+            Tags.AddOrUpdate(instrument.Name, _ => tags, (_, _) => tags);
+        });
         meterListener.Start();
     }
 
     public static TestingMetricListener SetupNServiceBusMetricsListener() =>
-        SetupMetricsListener("NServiceBus.Core");
+        SetupMetricsListener("NServiceBus.Core.Pipeline.Incoming");
```

Comment:
I am really happy with the direction the metrics implementation went in!  Great stuff.  For this change to listener name, I recommend mentioning this as a breaking change in https://github.com/Particular/NServiceBus/releases/tag/9.1.0 that if using OpenTelemetry, calls to AddMeter on MeterProviderBuilder will need to change from "NServiceBus.Core" to either "NServiceBus.Core.Pipeline.Incoming" or "NServiceBus.Core*".

@SzymonPobiega

Reply:
Thanks, @bbrandt. I started by [raising a docs PR to create a dedicated upgrade giode](https://github.com/Particular/docs.particular.net/pull/6774). ONce that's merged, I'll link to it from the relevant places.

Reply:
[Announcement](https://discuss.particular.net/t/nservicebus-9-1-0-minor-release-available/4030) and [release notes](https://github.com/Particular/NServiceBus/releases/tag/9.1.0) have been updated to link to the [newly created upgrade guide](https://docs.particular.net/nservicebus/upgrades/9to9.1/). @bbrandt, thanks again fro the feedback.